frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeaturesBatchActions/ArchiveButton.tsx

```diff
@@ -1,23 +1,52 @@
-import { useState, VFC } from 'react';
+import { useMemo, useState, VFC } from 'react';
 import { Button } from '@mui/material';
 import { PermissionHOC } from 'component/common/PermissionHOC/PermissionHOC';
 import { DELETE_FEATURE } from 'component/providers/AccessProvider/permissions';
 import useProject from 'hooks/api/getters/useProject/useProject';
 import { FeatureArchiveDialog } from 'component/common/FeatureArchiveDialog/FeatureArchiveDialog';
 import { usePlausibleTracker } from 'hooks/usePlausibleTracker';
+import { FeatureSchema } from 'openapi';
+import { addDays, isBefore } from 'date-fns';
 
 interface IArchiveButtonProps {
     projectId: string;
-    features: string[];
+    featureIds: string[];
+    features: FeatureSchema[];
 }
 
+const DEFAULT_USAGE_THRESHOLD_DAYS = 7;
+
+const useFeatureUsage = () => {
```

Comment:
why do we have useXYZ convention for read hooks. Isn't it a regular function? Also why do we have a function that creates another function? Can we drop the outermost function?

`const isFeatureInUser = (feature?: FeatureSchema) => ...`

Reply:
yup makes sense